jujugui/static/gui/src/app/app.js

```diff
@@ -1637,12 +1637,15 @@ YUI.add('juju-gui', function(Y) {
         console.log('switching environments is not supported in sandbox');
         return;
       }
-      // Set the credentials so the GUI will automatically log in when we
-      // switch the environments.
-      this.env.setCredentials({
-        user: username,
-        password: password
-      });
+      if (this.jem) {
+        // Set the credentials so the GUI will automatically log in when we
+        // switch the environments. We only switch if using JEM where user/pass
```

Comment:
Maybe " We only set credentials"?
Otherwise this is quite confusing, as we can also switch between models in the controller scenario without jem.